2010-11-10	Francisco Blas Izquierdo Riera <klondike@xiscosoft.es>

		(No bug)
		* compile: 			Separate LINK command and LINKFLAGS so libraries are searched properly.
		* documenter			Likewise.
		* makefile			Likewise.
		* sources			Likewise.


diff -ru -x define bpm2/compile bpm1/compile
--- bpm2/compile	2009-12-03 13:02:13.000000000 +0100
+++ bpm1/compile	2011-02-17 04:00:52.047625729 +0100
@@ -130,15 +130,15 @@
 
 bpmplay: $(BPMPLAY)
 	@echo " [link] "$@
-	@$(LINK) $(BPMPLAY) -o bpmplay
+	@$(LINK) $(BPMPLAY) -o bpmplay $(LINKFLAGS)
 
 bpmcount: $(BPMCOUNT_OBJECTS)
 	@echo " [link] "$@
-	@$(LINK) $(BPMCOUNT_OBJECTS) -o bpmcount
+	@$(LINK) $(BPMCOUNT_OBJECTS) -o bpmcount $(LINKFLAGS)
 
 bos: bos.o lock.o overseer.o process.o
 	@echo " [link] "$@
-	@$(LINK) $^ -o bos
+	@$(LINK) $^ -o bos $(LINKFLAGS)
 
 bpm.a:  $(BPM_LIB)
 	@echo "   [ar] "$@
@@ -146,7 +146,7 @@
 
 bpmdj: $(BPMDJ_OBJECTS) 
 	@echo " [link] "$@
-	@$(LINK) $(BPMDJ_OBJECTS) -o bpmdj
+	@$(LINK) $(BPMDJ_OBJECTS) -o bpmdj $(LINKFLAGS)
 
 profile-clock: profile-clock.o
 	@echo " [link] "$@
@@ -154,18 +154,18 @@
 
 bpmmerge: $(MERGER_OBJECTS)
 	@echo " [link] "$@
-	@$(LINK)  $(MERGER_OBJECTS) -o bpmmerge
+	@$(LINK)  $(MERGER_OBJECTS) -o bpmmerge $(LINKFLAGS)
 
 bpmclock: $(BPMCLOCK_OBJECTS) 
 	@echo " [link] "$@
-	@$(LINK)  $(BPMCLOCK_OBJECTS) -o bpmclock
+	@$(LINK)  $(BPMCLOCK_OBJECTS) -o bpmclock $(LINKFLAGS)
 
 idx2txt: idx2txt.o data.a
 	@echo " [link] "$@
-	@$(LINK)  $^ -o idx2txt
+	@$(LINK) $^ -o idx2txt $(LINKFLAGS)
 
 index-demo: index-demo.o index.o spectrum-type.o sample4-type.o power-type.o \
         memory.o data.a types.o common.o song-information.o capacity.o \
         song-information.moc.o info.o files.o
 	@echo " [link] "$@
-	@$(LINK)  $^ -o $@
+	@$(LINK)  $^ -o $@ $(LINKFLAGS)
diff -ru -x define bpm2/documenter bpm1/documenter
--- bpm2/documenter	2009-11-05 11:00:04.000000000 +0100
+++ bpm1/documenter	2011-02-17 03:31:07.643625727 +0100
@@ -12,7 +12,7 @@
 
 invert: invert.o
 	@echo " [link] "$@
-	@$(LINK) invert.o -o invert
+	@$(LINK) invert.o -o invert $(LINKFLAGS)
 
 %.o: %.cpp
 	@echo "  [cpp] "$@
diff -ru -x define bpm2/makefile bpm1/makefile
--- bpm2/makefile	2009-12-08 23:07:44.000000000 +0100
+++ bpm1/makefile	2011-02-17 03:30:21.511625729 +0100
@@ -14,7 +14,8 @@
 CFLAGS=-DLINUX -DSHORT_ONE
 JOBS=1
 include defines
-LINK =  $(CPP) $(LDFLAGS) $(QT_INCLUDE_PATH) $(QT_LIBRARY_PATH) $(QT_LIBS)
+LINK =  $(CPP) 
+LINKFLAGS = $(LDFLAGS) $(QT_INCLUDE_PATH) $(QT_LIBRARY_PATH) $(QT_LIBS)
 
 # Check all the link targets
 .link-targets:
diff -ru -x define bpm2/sources bpm1/sources
--- bpm2/sources	2009-12-04 13:16:00.000000000 +0100
+++ bpm1/sources	2011-02-17 03:32:04.222625727 +0100
@@ -44,20 +44,20 @@
 
 aoc: ao-syntax.o ao-lexer.o aoc.o sao-printer.o
 	@echo " [link] "$@
-	@$(LINK) $(CFLAGS) $^ -o $@
+	@$(LINK) $(CFLAGS) $^ -o $@  $(LINKFLAGS)
 
 
 doc: do-syntax.o do-lexer.o doc.o do-printer.o doc.o
 	@echo " [link] "$@
-	@$(LINK) $(CFLAGS) $^ -o $@		
+	@$(LINK) $(CFLAGS) $^ -o $@  $(LINKFLAGS)
 
 fixname: fixname.o
 	@echo " [link] "$@
-	@$(LINK) $(CFLAGS) $< -o $@
+	@$(LINK) $(CFLAGS) $< -o $@  $(LINKFLAGS)
 
 profile-clock: profile-clock.o
 	@echo " [link] "$@
-	@$(LINK) $(CFLAGS) $< -o $@
+	@$(LINK) $(CFLAGS) $< -o $@  $(LINKFLAGS)
 
 version.h: profile-clock
 	@echo " [test] clock - 10 seconds"
